TYPO3.Flow/Classes/TYPO3/Flow/Http/RequestHandler.php

```diff
@@ -129,7 +134,7 @@ public function handleRequest()
      */
     public function getHttpRequest()
     {
-        return $this->request;
+        return $this->componentContext->getHttpRequest();
```

Comment:
famous last words ;)
But as the null-check doesn't really solve the core issue, it might be worth to think about a completely different approach to the whole HTTP handling. So, I'd say: don't worry about the follow-up but lets collect some ideas for a more stable HTTP foundation (backwards compatible ofc)